packages/governance/src/types.js

```diff
@@ -1,77 +1,128 @@
  */
 
 /**
- * @typedef { Amount | Brand | Installation | Instance | bigint | Ratio | string | unknown } ParamValue
+ * @typedef { 'param_change' | 'election' | 'survey' } ElectionType
+ * param_change is very specific. Survey means multiple answers are possible,
+ * Election means some candidates are going to "win". It's not clear these are
+ * orthogonal. The important distinction is that param_change has a structured
+ * question, while the others have a question presented as a string.
  */
 
 /**
- * @typedef { 'choose_n' | 'order' | 'weight' } ChoiceMethod
+ * @typedef { 'half' | 'all' | 'none' } QuorumRule
  */
 
 /**
- * @typedef {Object} ParamDescription
- * @property {string} name
- * @property {ParamValue} value
- * @property {ParamType} type
+ * @typedef {Object} SimpleQuestion
+ * @property {string} question
  */
 
 /**
- * @typedef {Object} ParamManagerBase
- * @property {() => Record<Keyword,ParamDescription>} getParams
- *
- * @typedef {{ [updater: string]: (arg: ParamValue) => void }} ParamManagerUpdaters
- * @typedef {ParamManagerBase & ParamManagerUpdaters} ParamManagerFull
+ * @typedef { 'amount' | 'brand' | 'installation' | 'instance' | 'nat' | 'ratio' | 'string' | 'unknown' } ParamType
  */
 
 /**
- * @typedef {Array<ParamDescription>} ParamDescriptions
+ * @typedef { Amount | Brand | Installation | Instance | bigint | Ratio | string | unknown } ParamValue
  */
 
 /**
- * @callback BuildParamManager
- * @param {ParamDescriptions} paramDesc
- * @returns {ParamManagerFull}
+ * @typedef { SimpleQuestion | ParamChangeQuestion } Question
  */
 
 /**
- * @typedef {Object} QuestionTermsShort
- *   BallotDetails as provided to the Registrar
- * @property {ChoiceMethod} method
- * @property {string} question
+ * @typedef {Object} QuestionTerms - BallotSpec plus the Registrar Instance
+ * @property {Question} question
  * @property {string[]} positions
+ * @property {ChoiceMethod} method
  * @property {number} maxChoices
  * @property {ClosingRule} closingRule
+ * @property {Instance} registrar
+ */
+
+/**
+ * @typedef {Object} textPosition
+ * @property {string} text
  */
 
 /**
- * @typedef {Object} QuestionTerms
- *   BallotDetails after the Registrar adds its Instance
+ * @typedef { textPosition | changeParamPosition | noChangeParamPosition } Position
+ */
+
+/**
+ * @typedef {Object} BallotSpec
+ *   Specification when requesting a Ballot
  * @property {ChoiceMethod} method
- * @property {string} question
- * @property {string[]} positions
+ * @property {Question} question
+ * @property {Position[]} positions
+ * @property {ElectionType} electionType
  * @property {number} maxChoices
  * @property {ClosingRule} closingRule
- * @property {Instance} registrar
+ * @property {QuorumRule} quorumRule
+ * @property {Position} tieOutcome
  */
+
 /**
  * @typedef {Object} BallotDetails
- *   BallotDetails after the Registrar adds its Instance
+ *    complete ballot details: ballotSpec plus counter and handle
```

Comment:
I have the same maintenance hazard concerns here as with `QuestionTerms` above, conditional on the same question: Is `BallotDetails` really supposed to be `BallotSpec` + other stuff?